src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -251,9 +251,16 @@ let private VerifyEqualityComparison context addError (lhsType, lhsRange) (rhsTy
     // comparison for any derived type).
     let argumentError = ErrorCode.ArgumentMismatchInBinaryOp, [toString lhsType; toString rhsType]
     let baseType = CommonBaseType addError argumentError context.Symbols.Parent (lhsType, lhsRange) (rhsType, rhsRange)
+
+    // This assumes that:
+    // - Result has no derived types that support equality comparisons.
+    // - Compound types containing Result (e.g., tuples or arrays of results) do not support equality comparison.
```

Comment:
Q# currently doesn't support equality on tuple and array types, but if it does, the verification will not see it as a result comparison and will allow it. @bettinaheim Tuple/array equality seems useful, will Q# support it? Should this code handle that case?

Reply:
Yes, it would be nice to support it in the future, and it would be good if the code handles that.

Reply:
If you try to compare two types like this, should there only be one error (tuple/array equality not supported), or should there be an additional error from the capability verification? I feel like if equality is not supported on those types, there shouldn't be an additional error about result equality, but I'm not sure how to test the new code if the observable behavior is the same as the old code.

Reply:
Hm, as far as error messages go, it is good to spawn less errors rather than more; the errors should be actionable, and having a clear set of errors to fix rather than a list of 500 subsequent ones generally helps that I think.
However, since these are kind of two distinct things, my inclination here would be to indeed implement it and even give two errors (since the error is really about two different issues).

Reply:
OK, I'd like to do this in a separate PR - I added a task for it to #487.

Reply:
Ok, sounds good.